include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -1276,7 +1271,8 @@ class __brick_set_op
 
         auto __res = __internal::__pstl_lower_bound(__b, _Size2(0), __nb, __val_a, __comp);
 
-        bool bres = _IsOpDifference(); //initialization in true in case of difference operation; false - intersection.
+        constexpr bool __is_difference = std::is_same_v<_IsOpDifference, oneapi::dpl::unseq_backend::_DifferenceTag>;
+        bool bres = __is_difference; //initialization in true in case of difference operation; false - intersection.
```

Comment:
Small typo suggestion
```suggestion
        bool bres = __is_difference; //initialization is true in case of difference operation; false - intersection.
```